pyfixmsg/reference.py

```diff
@@ -175,7 +175,7 @@ def __init__(self, xml_file, eager=False):
         # this is effectively noop on python 2.7
         self.msg_types.update({m.msgtype: m for m in
                                (MessageType(e, self) for e in self.tree.findall('messages/message'))})
-        self.header_tags = [self.tags.by_name(t.get('name')) for t in self.tree.findall('header/field')]
+        self.header_tags = [self.tags.by_name(t.get('name')).tag for t in self.tree.findall('header/field')]
```

Comment:
It works actually. But I still have a concern. because some of our internal project also includes pyfixmsg as a module, so I'm afraid that change the data type of FixSpec.header_tags from [FixTag] to [int] might lead to some incompatibility issue. so I suggest that in function **_extract_sorting_key()**:

`
header_tags = spec.header_tags or [self.tags.by_name(tag) for tag in HEADER_TAGS]
for index, item in enumerate(header_tags):
    sorting_key[item.tag] = index
`

Reply:
you're right, fixed.